compiler/damlc/daml-stdlib-src/Prelude.daml

```diff
@@ -1,13 +1,19 @@
 -- Copyright (c) 2019 The DAML Authors. All rights reserved.
 -- SPDX-License-Identifier: Apache-2.0
-
+{-# LANGUAGE CPP #-}
 
 daml 1.2
 -- | The pieces that make up the DAML language.
 module Prelude (module X) where
 
 import DA.Internal.Prelude as X hiding (magic)
 import DA.Internal.LF as X hiding (Pair(..), TextMap, unpackPair)
+-- Template desugaring uses fromAnyTemplate and toAnyTemplate so we
+-- can’t remove them from the typeclass for older LF versions
+-- but we can hide them.
 import DA.Internal.Template as X
+#ifndef DAML_ANY_TEMPLATE
+  hiding (fromAnyTemplate, toAnyTemplate)
```

Comment:
@associahedron I’m not entirely satisfied with this but I don’t really know a better way given that template desugaring does not depend on the LF version.

Reply:
This seems fine. Nothing about using CPP is ever elegant, so...